### PR TITLE
Use cffi 1.17.0rc1 to test Python 3.13 support

### PR DESCRIPTION
**Pull request by edgarrmondragon**
## What do these changes do?

* Use a prerelease of `cffi` to support testing with Python 3.13

## Checklist

- [ ] I think the code is well written
- [ ] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [ ] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names.
- [ ] Add a new news fragment into the `CHANGES/` folder
  * name it `<issue_or_pr_num>.<type>.rst` (e.g. `588.bugfix.rst`)
  * if you don't have an issue number, change it to the pull request
    number after creating the PR
    * `.bugfix`: A bug fix for something the maintainers deemed an
      improper undesired behavior that got corrected to match
      pre-agreed expectations.
    * `.feature`: A new behavior, public APIs. That sort of stuff.
    * `.deprecation`: A declaration of future API removals and breaking
      changes in behavior.
    * `.breaking`: When something public is removed in a breaking way.
      Could be deprecated in an earlier release.
    * `.doc`: Notable updates to the documentation structure or build
      process.
    * `.packaging`: Notes for downstreams about unobvious side effects
      and tooling. Changes in the test invocation considerations and
      runtime assumptions.
    * `.contrib`: Stuff that affects the contributor experience. e.g.
      Running tests, building the docs, setting up the development
      environment.
    * `.misc`: Changes that are hard to assign to any of the above
      categories.
  * Make sure to use full sentences with correct case and punctuation,
    for example:
    ```rst
    Fixed issue with non-ascii contents in doctest text files
    -- by :user:`contributor-gh-handle`.
    ```

    Use the past tense or the present tense a non-imperative mood,
    referring to what's changed compared to the last released version
    of this project.

### CODE REVIEW

File: .github/workflows/ci-cd.yml [hunk 1]
```diff
@@ -189,7 +189,7 @@ jobs:
             pip-ci-${{ runner.os }}-${{ matrix.pyver }}-${{ matrix.no-extensions }}-
     - name: Update pip, wheel, setuptools, build, twine
       run: |
-        python -m pip install -U pip wheel setuptools build twine
+        python -m pip install -U pip wheel setuptools build twine cffi==0.17.0rc1
```
Comment (webknjaz) [inline .github/workflows/ci-cd.yml]: This seems redundant as the following step would do the same.

Comment (edgarrmondragon) [reply .github/workflows/ci-cd.yml]: Yeah, but `pip` (I think) has a dependency on cffi via cryptography so this step was failing on 3.13

**Comment by edgarrmondragon** [comment]
It seems [pycares](https://github.com/saghul/pycares) is also causing issues

File: .github/workflows/ci-cd.yml (hunk 1 above)
Comment (webknjaz) [reply .github/workflows/ci-cd.yml]: Pip vendors all its deps so this would do nothing. Besides, all pip's deps are pure-python.

**Comment by Dreamsorcerer** [comment]
Looks like uvloop is also failing to install currently. I suspect there's a good chance cffi will have made a stable release by the time that's sorted, so maybe not worth putting these changes in temporarily.

**Comment by bdraco** [comment]
Looks like cffi 1.17.0 is now out

**Comment by Dreamsorcerer** [comment]
I guess we close this then, as Dependabot will update it. Just waiting on uvloop now..

**Pull request closed without merging**